### PR TITLE
Auto-breakout - Module implementation

### PR DESCRIPTION
**Pull request by SaiTeja0510**
##### SUMMARY
Auto-breakout - Module implementation

##### GitHub Issues

| GitHub Issue # |
| -------------- |
| N/A|

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

sonic_system

##### ADDITIONAL INFORMATION

```paste below
N/A
```

##### Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [X] I have not allowed coverage numbers to degenerate
- [] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have maintained backward compatibility or have provided any relevant "breaking_changes" descriptions in a "fragment" file in the "changelogs/fragments" directory of this repository.

##### How Has This Been Tested?

- [X] Regression test HTML report:
[sonic_system_test_results.pdf](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/14372269/sonic_system_test_results.pdf)
- [X] Regression test detailed terminal log:
[system_autobreakout__UT.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/14372275/system_autobreakout__UT.txt)
- [X] facts_UT log:
[system_autobreakout_facts_UT.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/14372290/system_autobreakout_facts_UT.txt)

Edit 1:

Adding new UT logs which includes lates changes of system module.
[system_autobreakout_UT_2.txt](https://github.com/ansible-collections/dellemc.enterprise_sonic/files/15095460/system_autobreakout_UT_2.txt)

### CODE REVIEW

**Comment by kerry-meyer** [comment]
Thank you for providing these changes to support the Auto-breakout configuration attribute.

In order to proceed with the code review for this PR, please do the following:

- Update the "module" file for the "system" resource module (plugins/modules/sonic_system.py) to document the new "auto_breakout" configuration attribute. (Only a sentence or two is needed.)
- Fix the (white space) sanity failures for the PR contents. REF: "Details" link for any failing CI/sanity check for the PR (near the bottom of the "Conversation" tab of the PR) or https://github.com/ansible-collections/dellemc.enterprise_sonic/actions/runs/8002707024/job/21856717555?pr=342
- Add UT test cases that will cause the new code to be executed and tested during UT execution. (Changes are needed in the following files to add this UT coverage: "tests/unit/modules/network/sonic/test_sonic_system.py" and "tests/unit/modules/network/sonic/fixtures/sonic_system.yaml".)

**Comment by SaiTeja0510** [comment]
> Thank you for providing these changes to support the Auto-breakout configuration attribute.
> 
> In order to proceed with the code review for this PR, please do the following:
> 
> * Update the "module" file for the "system" resource module (plugins/modules/sonic_system.py) to document the new "auto_breakout" configuration attribute. (Only a sentence or two is needed.)
> * Add a "fragment" file to describe this change in the "changelogs/fragments" file for use in the upcoming release notes. (See other files in the target directory for examples for format and content of this one-line file.)
> * Fix the (white space) sanity failures for the PR contents. REF: "Details" link for any failing CI/sanity check for the PR (near the bottom of the "Conversation" tab of the PR) or https://github.com/ansible-collections/dellemc.enterprise_sonic/actions/runs/8002707024/job/21856717555?pr=342
> * Add UT test cases that will cause the new code to be executed and tested during UT execution. (Changes are needed in the following files to add this UT coverage: "tests/unit/modules/network/sonic/test_sonic_system.py" and "tests/unit/modules/network/sonic/fixtures/sonic_system.yaml".)

Thank you for the feedback.
I have addressed the comments. Please review and let me know if anything else needs to me changed.

File: plugins/module_utils/network/sonic/config/system/system.py [hunk 1]
```diff
@@ -395,6 +411,7 @@ def get_replaced_config(self, have, want):
 
         w_hostname = want.get('hostname', None)
         w_intf_name = want.get('interface_naming', None)
+        w_auto_breakout_mode = want.get('auto_breakout', None)
```
Comment (mingjunzhang2019) [inline plugins/module_utils/network/sonic/config/system/system.py:414]: Should line#416 be added with "or  w_auto_breakout_mode is not None"?

File: plugins/modules/sonic_system.py [hunk 2]
```diff
@@ -75,6 +75,13 @@
             description:
               - Specifies the mac anycast-address
             type: str
+      auto_breakout:
+        description:
+          - Specifies auto-breakout status in the device
+        type: str
+        choices:
+          - ENABLE
+          - DISABLE
```
Comment (mingjunzhang2019) [inline plugins/modules/sonic_system.py:84]: The following "EXAMPLES" section should be added with some examples for auto_breakout, which should come from the resource model examples files.

File: plugins/module_utils/network/sonic/facts/system/system.py [hunk 3]
```diff
@@ -86,6 +86,22 @@ def get_anycast_addr(self):
...
+            response = edit_config(self._module, to_request(self._module, request))
+        except ConnectionError as exc:
+            self._module.fail_json(msg=str(exc), code=exc.code)
+        if ('sonic-device-metadata:DEVICE_METADATA_LIST' in response[0][1]):
+            auto_breakout_data = response[0][1]['sonic-device-metadata:DEVICE_METADATA_LIST']
+            if 'auto-breakout' in auto_breakout_data[0]:
+                auto_breakout_val = auto_breakout_data[0]['auto-breakout']
+                data = {'auto-breakout': auto_breakout_val}
+            else:
+                data = {}
```
Comment (mingjunzhang2019) [inline plugins/module_utils/network/sonic/facts/system/system.py:102]: If line#96 is false through, data will not have definition. So data can be initiated before line#96 and remove line#101 and 102.

File: plugins/module_utils/network/sonic/config/system/system.py (hunk 1 above)
Comment (SaiTeja0510) [reply plugins/module_utils/network/sonic/config/system/system.py:414]: > Should line#416 be added with "or w_auto_breakout_mode is not None"?

Yes that had to be added. Addressed it. Thank you for pointing it out.

File: plugins/modules/sonic_system.py (hunk 2 above)
Comment (SaiTeja0510) [reply plugins/modules/sonic_system.py:84]: > The following "EXAMPLES" section should be added with some examples for auto_breakout, which should come from the resource model examples files.

Added examples in the file.

File: plugins/module_utils/network/sonic/facts/system/system.py (hunk 3 above)
Comment (SaiTeja0510) [reply plugins/module_utils/network/sonic/facts/system/system.py]: > If line#96 is false-through, the "data" will not have definition. The "data" can be initiated before line#96 and remove line#101 and 102.

Agreed. Updated the code.

**Review by mingjunzhang2019 — APPROVED** [review]

**Review by kerry-meyer — APPROVED** [review]
The proposed code changes and posted test results look good.

Thank you for providing this enhancement to the functionality of the Dell enterprise_sonic Ansible resource module collection.

Approved.

**Pull request merged**